assets/js/googlesitekit/widgets/components/WidgetWrapper.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * WidgetWrapper component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { HIDDEN_CLASS } from '../util/constants';
+import { Cell } from '../../../material-components';
+
+const WidgetWrapper = ( { gridColumnWidth, children } ) => {
+	if ( 0 === gridColumnWidth ) {
+		// Widget with columnWidth of 0 should be hidden
+		return (
+			<div className={ HIDDEN_CLASS }>
+				{ children }
+			</div>
+		);
+	}
+
+	if ( gridColumnWidth === 3 || gridColumnWidth === 4 ) {
+		return (
+			<Cell lgSize={ gridColumnWidth } mdSize={ 4 } smSize={ 2 }>
+				{ children }
+			</Cell>
+		);
+	}
+
+	if ( gridColumnWidth === 6 || gridColumnWidth === 8 ) {
+		return (
+			<Cell lgSize={ gridColumnWidth } mdSize={ 8 }>
+				{ children }
+			</Cell>
+		);
+	}
```

Comment:
This doesn't fully match the AC requirements:
1. The extra rule for `6` and `8` shouldn't be needed; in that case it should simply be a single `size`.
2. The rule for `3` and `4` shouldn't be that specific. While the ACs mention that these are realistically the only values that would be used, it states that this should apply to any width lower than 6 (except 0 of course).

So basically, we can remove one condition here completely while making the other apply to anything below 6. Unless there was a specific reason you diverged from the ACs here?

Reply:
As I explained below, I tried to use the snapshots as the source of truth throughout. 

> The extra rule for 6 and 8 shouldn't be needed; in that case it should simply be a single size.

In this case I deviated from the ACs because I was seeing diffs like this:
### 
```diff
-  class="mdc-layout-grid__cell mdc-layout-grid__cell--span-6-desktop mdc-layout-grid__cell--span-8-tablet"
+  class="mdc-layout-grid__cell mdc-layout-grid__cell--span-6"
```
I was concerned that this might cause a regression. It seems that for a `6` and `8` on desktop, we currently do `8` on tablet. Can you advise?

> The rule for 3 and 4 shouldn't be that specific. While the ACs mention that these are realistically the only values that would be used, it states that this should apply to any width lower than 6 (except 0 of course).

I can certainly change this 👍

Reply:
@johnPhillips Ah right, thanks, that's a good catch. In that case, it should be adjusted as follows:
* If smaller than 6, use the first condition (4 on medium, 2 on small)
* If smaller than 8, use the second condition (8 on medium)
* Otherwise use the default (no fallback, always full amount of columns)

Reply:
No problem, thanks!